lib/evss/disability_compensation_form/form0781.rb

```diff
@@ -12,6 +12,7 @@ def initialize(user, form_content)
         @user = user
         @phone_email = form_content.dig('form526', 'phoneAndEmail')
         @form_content = form_content.dig('form526', 'form0781')
+        @sync_modern_0781_flow = form_content.dig('form526', 'syncModern0781Flow')
```

Comment:
Are we not able to determine based on the structure such that we need to add this key to the overall 526 submission data? Like that `'form0781v2'` key from above?

Reply:
Do you mean the `'form0781'` key on L14?

Within `'form0781'`, I wasn't able to determine a better attribute that would identify the form content contains data corresponding to the new 0781 form. The `'syncModern0781Flow'` attribute will be submitted as part of the `'form526'` object, regardless, as it persists throughout the form flow to determine which pages are rendered. Therefore, I figured that was as good an indicator as any with which to determine form translation.

Would be happy to entertain any suggestions you might have, however.

Reply:
You are doing

`form_content = parsed_forms[form_key]`

`form_key` can equal:

- 'form0781' 
- 'form0781a'
- 'form0781v2' 

This implies, there is a `submission.form['form0781']['form0781v2']` in the submission data, when there is new form data. 

Unless I am misunderstanding? I would think, you key off that, then dont need an extra key. The data/key is either there, or not. 

Again, I may be missing something but that was my thought process.

Reply:
Those 3 keys, 'form0781', 'form0781a' and 'form0781v2', are created in the `translate` method below. 

`form_content` is what is submitted by the FE, which doesn't currently break out 'form0781' and 'form0781a' data by key; it's different that what is transformed here and saved to a submission record. 

The 'form0781' and 'form0781a' data come nested under the 'form0781' key, and are differentiated below in the `split_incidents` method, depending on the value of the `'personalAssault'` attribute for each `incident'`.

The FE-submitted data and JSON schema could be modified to contain all new 0781 data under a 'form0781v2' key, but I thought the approach was to follow the current process.

Let me know your thoughts. Thanks.